napari/utils/io.py

```diff
@@ -127,3 +128,25 @@ def read_zarr_dataset(path):
     else:
         raise ValueError(f"Not a zarr dataset or group: {path}")
     return image, shape
+
+
+def write_csv(path, data, column_names=None):
```

Comment:
The files and pandas dataframes are the same **if you set the index column equal to "coord_id"**.

```python
table, column_names = my_points.to_table(path='my_file_1.csv')
points_dataframe = pandas.DataFrame(table, columns=column_names).set_index('coord_id')
points_dataframe.to_csv('my_file_2.csv')
my_new_dataframe = pandas.read_csv('my_file_2.csv', index_col='coord_id') 
```

Result:
* 'my_file_1.csv' and 'my_file_2.csv' have the same contents.
*  `np.allclose(points_dataframe, my_new_dataframe)` returns True.

The downside is, it's hard to remember to explicitly set the index of the pandas dataframe. But I really don't want to write out the useless pandas index to file, so I think we'll just have to live with it. 

We can add a file to the examples that describes essentially the same thing as this comment, if you like.

Reply:
Why do we have a thing called `coord_id`? I'd prefer an index-less dataframe without point ids...

Reply:
I did that for consistency with the shapes csv, although I'm open to dropping that column.

I think we wanted to keep both the 'shape_id' and 'coord_id' for the shapes csv, particularly because the order of coordinates in a shape is important.

### Shapes csv
What we have now, what we plan to keep:

| shape_id | shape_type | coord_id | dim_0 | dim_1 |  dim_2 | ... | dim_n |
|----------|------------|----------|-------|-------|--------|-----|-------|
| 0        | Ellipse    | 0        |       |       |        |     |       |
| 0        | Ellipse    | 1        |       |       |        |     |       |
| 0        | Ellipse    | 2        |       |       |        |     |       |
| 0        | Ellipse    | 3        |       |       |        |     |       |
| 1        | Rectangle  | 0        |       |       |        |     |       |
| 1        | Rectangle  | 1        |       |       |        |     |       |
| ...      | ...        | ...      |       |       |        |     |       |
| n        | Path       | n-1      |       |       |        |     |       |
| n        | Path       | n        |       |       |        |     |       |

### Points csv
What we have now:

| coord_id | dim_0 | dim_1 | ... | dim_n |
|----------|-------|-------|-----|-------|
| 0        |       |       |     |       |
| 1        |       |       |     |       |
| ...      |       |       |     |       |

What @jni would like instead:

| dim_0 | dim_1 | ... | dim_n |
|-------|-------|-----|-------|
| .      |       |     |       |
| .      |       |     |       |
| .      |       |     |       |